numpy/core/tests/test_simd.py

```diff
@@ -330,12 +330,15 @@ def test_square(self):
         square = self.square(vdata)
         assert square == data_square
 
-    @pytest.mark.parametrize("intrin, func", [("self.ceil", math.ceil)])
+    @pytest.mark.parametrize("intrin, func", [("self.ceil", math.ceil),
+    ("self.ceil", math.ceil)])
```

Comment:
typo I guess :)

Reply:
you must ship it to python too through [_simd.dispatch.c.src](https://github.com/numpy/numpy/blob/main/numpy/core/src/_simd/_simd.dispatch.c.src)

Reply:
added

Reply:
But I am meeting a weird situation now. If I understand it correctly, the addition and subtraction of the magic number 2^52 is the same as truncation. I ran the same implementation on my local machine, and I worked fine. However, the implementation became `ceil` in CI.

Reply:
> the addition and subtraction of the magic number 2^52 is the same as truncation

no, the idea behind it is to remove the fraction and let FPU round it according to MXCSR control mode which by default is set to "round to nearest, ties to even".

so if we assume MXCSR set to the default (Python demand it), then 
```Python
truncate = lambda x: math.copysign(abs(x) - (1.0 if round(x) > abs(x) else 0.0), x)
```

Reply:
according to your last commit, I think there's a miss understanding here. the example I have provided is just a pseudo-code of what you suppose to implement for `npyv_trunc_f64` on `sse2`.

Reply:
Thank you for the information. I have fixed the implementation!